packages/interpretations/src/models/interpretation.js

```diff
@@ -24,42 +25,40 @@ export default class Interpretation {
         const modelId = this._parent.id;
         const modelName = this._parent.modelDefinition.name;
         const isNewInterpretation = !this.id;
-        
+
         if (isNewInterpretation) {
             // Set initial sharing of interpretation from the parent object
-            
-            return await apiFetchWithResponse(d2, `/interpretations/${modelName}/${modelId}`, "POST", this.text)
-                .then(getInterpretationIdFromResponse)
-                .then(interpretationId => {
-                    this.id = interpretationId;
-                    const sharingPayload = this.sharing 
-                        ? { object: {...this.sharing, id: this.id} } 
-                        : { object: pick(Interpretation.sharingFields, this._parent) };
+            const response = await apiFetchWithResponse(d2, `/interpretations/${modelName}/${modelId}`, "POST", this.text);
+            const interpretationId = getInterpretationIdFromResponse(response);
+            const sharingUrl = `/sharing?type=interpretation&id=${interpretationId}`;
+
+            const sharingPayload = this.sharing
+                ? { object: { ...this.sharing, id: this.id } }
+                : { object: pick(Interpretation.sharingFields, this._parent) };
 
-                    this.sharing = null;
-                    const sharingUrl = `/sharing?type=interpretation&id=${interpretationId}`;
-                    return apiFetch(d2, sharingUrl, "PUT", sharingPayload).then(() =>  this);
-                });
+            this.sharing = null;
+            this.id = interpretationId;
+
+            return apiFetch(d2, sharingUrl, "PUT", sharingPayload)
         } else {
-            return await apiFetch(d2, `/interpretations/${this.id}`, "PUT", this.text)
-                .then(() => {
-                    if (this.sharing) {
-                        const sharingPayload =  { object: {...this.sharing, id: this.id}};
-                        this.sharing = null;
+            // interpretation already exists in DB
+            await apiFetch(d2, `/interpretations/${this.id}`, "PUT", this.text);
```

Comment:
Is the purpose of this fetch to determine if the interpretation exists?

Reply:
I assume it is for updating the text of interpretation/comment.